python/UniverseWrapper.cpp

```diff
@@ -471,6 +484,16 @@ namespace FreeOrionPython {
                                                     return_value_policy<return_by_value>()
                                                 ))
 
+            // Indexed by stat name (string), contains a map indexed by empire id,
+            // contains a map from turn number (int) to stat value (double).
+            .def("statRecords",                 make_function(
```

Comment:
It looks like a huge pile of information.  I don't like to work with large nested dicts because it is hard to understand them.  

Maybe we should split them by smaller parts. (by requesting only one meter or only for one empire.)

Can someone provide me a usecase?

Reply:
>  I don't like to work with large nested dicts because it is hard to understand them.

I think it is perfectly reasonable to provide a single interface function to get that "huge pile of information" as long as adequate python functions are used to access them.

There is no need to provide a large number of restricted C++ API interface functions in advance if you can just implement some python interface functions on a per need basis.

Reply:
This is just providing access to the data. I you wanted to do something with it, you'd probably extract the relevant slices by suitably indexing the multi-dimensional container. I'd rather not add a complicated interface for doing that indexing to the API when you can probably do so from within Python and will end up re-sorting all the results into Python containers anyway...

The use case that motivated this was accessing game statistics for exporting to track and compare AI performance: https://github.com/freeorion/freeorion/issues/2899